posts/renv-complications/index.qmd

```diff
@@ -0,0 +1,131 @@
+---
+title: "Things that can go wrong when using renv"
+author:
+  - name: "Hugo Gruson"
+    orcid: "0000-0002-4094-1476"
+date: "2023-09-26"
+categories: [R, reproducibility, renv]
+format:
+  html: 
+    toc: true
+---
+
+We use the [renv R package](https://rstudio.github.io/renv/) to ensure reproducibility of the training materials and the pipelines we are providing.
+But we sometimes get reports from users who struggle to rebuild the environment and run the code.
+
+In this post, we dissect the source of these issues, explain why in reality renv is not at fault, and how this is caused by the inherent complexity of reproducibility. renv documentation already includes [caveats](https://rstudio.github.io/renv/articles/renv.html#caveats) explaining why some situations are bound to require more complex tools. This blog post reiterates some of these caveats and illustrates them with concrete examples.
+
+We finish by mentioning a couple of more complete (but more complex!) frameworks that can overcome the issues presented here.
+We do not explore these alternative framework in details but we provide links to more information.
+
+## Binaries vs building from source
+
+Software, including R packages, can generally be delivered in two forms: as binaries or as source code.
+If you are installing as source code, you may in some case need a compilation toolchain on your computer.
+If the toolchain is missing, it can lead to errors such as:
```

Comment:
```suggestion
Software, including R packages, can generally be delivered in two forms: as binaries or as source code.
If you are building from the source code, you may in some case need a compilation toolchain on your computer.
If that toolchain is missing, it can lead to errors such as:
```

Consistency with the header, to prevent confusion.